File: Task-4/sharedArea.py

lowerLeft = [1,1]
upperRight = [1,1]
def main():
    test_cases = int(input())
    for testCase in range(test_cases):
        rectangles = int(input())
        x1,y1,x2,y2 = [int(i) for i in input().split()]
        global lowerLeft
        global upperRight
        lowerLeft = [x1,y1]
        upperRight = [x2,y2]
        for rectangle in range(rectangles-1):
            x3,y3,x4,y4 = [int(i) for i in input().split()]
            getSharedPoints(lowerLeft[0],lowerLeft[1],upperRight[0],upperRight[1],x3,y3,x4,y4)
        if upperRight[0]>lowerLeft[0] and upperRight[1]>lowerLeft[1]:
            print(f"Case #{testCase+1}: {getArea(lowerLeft,upperRight)}")
        else:
            print(f"Case #{testCase+1}: 0")


def getSharedPoints(x1,y1,x2,y2,x3,y3,x4,y4):
    max_x_bottomLeft = max(x1,x3)
    max_y_bottomLeft = max(y1,y3)
    min_x_upperRight = min(x2,x4)
    min_y_upperRight = min(y2,y4)
    global lowerLeft
    global upperRight
    lowerLeft = [max_x_bottomLeft,max_y_bottomLeft]
    upperRight = [min_x_upperRight,min_y_upperRight]



def getArea(lowerLeft,upperRight):
    height = upperRight[1]-lowerLeft[1]
    width = upperRight[0]-lowerLeft[0]
    return width*height

main()

File: Task-4/sharedArea.cpp
#include <iostream>
using namespace std;
void getSharedPoints(long long lowerLeft[], long long upperRight[], long long x3, long long y3, long long x4, long long y4);
long long getArea(long long lowerLeft[], long long upperRight[]);
int main()
{
    long long test_cases;
    cin >> test_cases;
    for (long long i = 0; i < test_cases; i++)
    {
        long long rects;
        cin >> rects;
        long long x1, y1, x2, y2;
        cin >> x1 >> y1 >> x2 >> y2;
        long long lowerLeft[] = {x1, y1};
        long long upperRight[] = {x2, y2};

        for (long long j = 0; j < rects - 1; j++)
        {
            long long x3, y3, x4, y4;
            cin >> x3 >> y3 >> x4 >> y4;
            getSharedPoints(lowerLeft, upperRight, x3, y3, x4, y4);
        }
        if (upperRight[0] > lowerLeft[0] && upperRight[1] > lowerLeft[1])
        {

            cout << "Case #" <<i+1 << ": " << getArea(lowerLeft, upperRight)<<endl;
        }
        else
        {
            cout << "Case #" <<i+1 << ": 0"<<endl;
        }
    }

    return 0;
}
void getSharedPoints(long long lowerLeft[], long long upperRight[], long long x3, long long y3, long long x4, long long y4)
{
    lowerLeft[0] = max(lowerLeft[0], x3);
    lowerLeft[1] = max(lowerLeft[1], y3);
    upperRight[0] = min(upperRight[0], x4);
    upperRight[1] = min(upperRight[1], y4);
}
long long getArea(long long lowerLeft[], long long upperRight[])
{
    return (upperRight[0] - lowerLeft[0]) * (upperRight[1] - lowerLeft[1]);
}